source/material_model/rheology/elasticity.cc

```diff
@@ -252,11 +252,33 @@ namespace aspect
             for (unsigned int i=0; i < in.n_evaluation_points(); ++i)
               {
                 // Get old stresses from compositional fields
+                // TODO: use $tau^{t+\Delta te}$ or evaluate the old solution for stress
+                // (and viscosity)
+                // Explanation: for the assembly of the Stokes equations (the force term goes
+                // into the rhs of the momentum equation), this function is evaluated.
+                // This is after the advection equations have been solved, and hence in.composition
+                // contains the rotated, averaged and advected stresses at $tau^{t+\Delta t}$.
+                //
+                // For an iterative Advection scheme, in.composition could even be different
+                // every time the Stokes system is assembled.
+                //
+                // Moresi et al. (2003) use the full stresses at $t$ (Eq.  30) in the force term.
+                // There is also an incorrect minus sign in the force term.
+                // The force term should be computed as:
+                // $\frac{-\eta_{effcreep}  tau_0}{\eta_{e}}$, where $\eta_{effcreep}$ is the
+                // harmonic average of the vp and e viscosity.
+                // How will we get to this viscosity? 
```

Comment:
That is even trickier if the viscosity we need to use is not just old the VEP or VE viscosity, but rather the harmonic average of all the different deformation mechanism viscosities. Actually, I think this should only be done if we need to locally iterate between the plastic and elastic stresses. If we use the standard viscosity rescaling method, then the old VEP viscosity should be correct.

To get the old viscosity, as discussed the options are:
1. Reevalaute the material model using the "old" values for the material model inputs
2. Store and keep track of the old viscosity in a compositional field.

In principal, I am fine with either solution.

Reply:
These are the viscosities given the strain rate at t+delta t_e. So the question for me is how we obtain the strain rate at this time, given that we only have the strain rate at t+delta t. Do we just assume that the strain rate is constant between t and t + delta t_e? This would be inconsistent with both the time-stepping, and with the assumption that the spin and rate of change of stress are both constant between t and t + delta t_e.

Reply:
We could extrapolate the strain rate linearly, which would imply that viscosities are independent of strain rate at each time step. That would be inconsistent with the rheological laws though.

Reply:
I'm tempted to suggest that we concentrate on fixing the Delta t_e = Delta t case, and add a warning for people attempting to vary the elastic timestep.